pandas/tests/groupby/test_categorical.py

```diff
@@ -453,6 +453,44 @@ def test_dataframe_categorical_with_nan(observed):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize("ordered", [True, False])
+@pytest.mark.parametrize("observed", [True, False])
+@pytest.mark.parametrize("sort", [True, False])
+def test_dataframe_categorical_ordered_observed_sort(ordered, observed, sort):
+    # GH 25871: Fix groupby sorting on ordered Categoricals
+    # Build a dataframe with cat having one unobserved category ('AWOL'),
+    # and a Series with identical values
+    cat = pd.Categorical(['d', 'a', 'b', 'a', 'd', 'b'],
+                         categories=['a', 'b', 'AWOL', 'd'],
+                         ordered=ordered)
+    val = pd.Series(['d', 'a', 'b', 'a', 'd', 'b'])
+    df = pd.DataFrame({'cat': cat, 'val': val})
+
+    # aggregate on the Categorical
+    result = (df.groupby('cat', observed=observed, sort=sort)['val']
+                .aggregate('first'))
+
+    # If ordering works, we expect index labels equal to aggregation results,
+    # except for 'observed=False': index contains 'AWOL' and aggregation None
+    label = pd.Series(result.index.array, dtype='object')
+    aggr = pd.Series(result.array)
+    if not observed:
+        aggr[aggr.isna()] = 'AWOL'
+    tm.assert_series_equal(label, aggr)
+
+    # GH 25167: Groupby with observed=True doesn't sort
```

Comment:
this test doesn't use any of the parameters? can you update that. you *may* want to make it a new test (so don't have to jump thru hoops to create the expected result)

Reply:
Can do. If I fully parameterise this test, it will overlap considerably with my new one, though, especially if I rewrite it along the lines of your other comment. As it is now, it's exactly the issue's sample code, thus proving that it's been solved.

But you can never have too many tests. I'll make it a separate one.

Reply:
really what am after here, IS this other test actually different? if not, then remove it and just add the issue number with the other. I agree can't have too many tests, but at the same times we ideally want *unique* tests.

Reply:
I agree. Will remove the second test.

Reply:
Removed in latest commit